app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java

```diff
@@ -41,32 +33,33 @@
     @BindView(R.id.contributionProgress) ProgressBar progressView;
     @BindView(R.id.failed_image_options) LinearLayout failedImageOptions;
 
-    @Inject
-    MediaDataExtractor mediaDataExtractor;
-
-    @Inject
-    @Named("thumbnail-cache")
-    LruCache<String, String> thumbnailCache;
-
-    private DisplayableContribution contribution;
-    private CompositeDisposable compositeDisposable = new CompositeDisposable();
     private int position;
+    private Contribution contribution;
 
     ContributionViewHolder(View parent, Callback callback) {
         super(parent);
         ButterKnife.bind(this, parent);
         this.callback=callback;
     }
 
-    public void init(int position, DisplayableContribution contribution) {
-        ApplicationlessInjection.getInstance(itemView.getContext())
-                .getCommonsApplicationComponent().inject(this);
+    public void init(int position, Contribution contribution) {
+        this.contribution=contribution;
         this.position=position;
-        this.contribution = contribution;
-        fetchAndDisplayThumbnail(contribution);
+        String imageSource = contribution.thumbUrl;
+        if (TextUtils.isEmpty(imageSource)) {
+            imageSource =
+                contribution.getLocalUri() != null ? contribution.getLocalUri().toString()
+                    : null;
+        }
+        if (!TextUtils.isEmpty(imageSource)) {
+            final ImageRequest imageRequest =
+                ImageRequestBuilder.newBuilderWithSource(Uri.parse(imageSource)).setProgressiveRenderingEnabled(true).setProgressiveRenderingEnabled(true)
```

Comment:
`.setProgressiveRenderingEnabled(true).setProgressiveRenderingEnabled(true)` so nice you said it twice